### ISSUE TITLE
[Android] Some System.Net.Security tests are failing

### ISSUE THREAD

**Issue by simonrozsival** [dotnet/runtime#68206]
The following `System.Net.Security` tests are failing on Android:
- CertificateValidationClientServer.CertificateSelectionCallback_DelayedCertificate_OK
- ServerAsyncAuthenticateTest.ServerAsyncAuthenticate_MismatchProtocols_Fails
- SslStreamAlpnTest_Async.SslStream_StreamToStream_Alpn_NonMatchingProtocols_Fail
- SslStreamAlpnTest_Sync.SslStream_StreamToStream_Alpn_NonMatchingProtocols_Fail
- SslStreamNetworkStreamTest.SslStream_TargetHostName_Succeeds
- SslStreamNetworkStreamTest.SslStream_ServerUntrustedCaWithCustomTrust_OK
- SslStreamNetworkStreamTest.SslStream_ClientCertificate_SendsChain
- SslStreamNetworkStreamTest.SslStream_ClientCertificateContext_SendsChain
- SslStreamSniTest.SslStream_ClientSendsSNIServerReceives_Ok
- SslStreamSniTest.SslStream_ServerCallbackNotSet_UsesLocalCertificateSelection
- SslStreamSniTest.UnencodedHostName_ValidatesCertificate
- TransportContextTest.TransportContext_ConnectToServerWithSsl_GetExpectedChannelBindings

**Comment by maraf** [comment]
Also `System.Net.Security.Tests.CertificateValidationRemoteServer.ConnectWithRevocation_WithCallback` from [#20220604.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1806758&view=logs&j=b6e5d97f-d4f8-506b-9e47-e3e32385754a&t=94ce2805-f3e2-5f9a-1d92-0da5613f733d)

```
System.Security.Authentication.AuthenticationException : Authentication failed, see inner exception.
---- Interop+AndroidCrypto+SslException : Exception of type 'Interop+AndroidCrypto+SslException' was thrown.
   at System.Net.Security.SslStream.<ForceAuthenticationAsync>d__144`1[[System.Net.Security.AsyncReadWriteAdapter, System.Net.Security, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].MoveNext()
   at System.Threading.Tasks.TaskTimeoutExtensions.GetRealException(Task task) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 120
--- End of stack trace from previous location ---
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 90
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks, Int32 millisecondsTimeout) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 55
   at System.Net.Security.Tests.CertificateValidationRemoteServer.ConnectWithRevocation_WithCallback_Core(X509RevocationMode revocationMode, Nullable`1 offlineContext) in /_/src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs:line 198
   at System.Net.Security.Tests.CertificateValidationRemoteServer.ConnectWithRevocation_WithCallback_Core(X509RevocationMode revocationMode, Nullable`1 offlineContext) in /_/src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs:line 199
   at System.Net.Security.Tests.CertificateValidationRemoteServer.ConnectWithRevocation_WithCallback_Core(X509RevocationMode revocationMode, Nullable`1 offlineContext) in /_/src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs:line 199
   at System.Net.Security.Tests.CertificateValidationRemoteServer.ConnectWithRevocation_WithCallback_Core(X509RevocationMode revocationMode, Nullable`1 offlineContext) in /_/src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs:line 199
   at System.Net.Security.Tests.CertificateValidationRemoteServer.ConnectWithRevocation_WithCallback_Core(X509RevocationMode revocationMode, Nullable`1 offlineContext) in /_/src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs:line 199
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
```

**Comment by maraf** [comment]
cc @karelz

**Comment by maraf** [comment]
Moved to a separate issue as dug-out more details https://github.com/dotnet/runtime/issues/70469

**Comment by simonrozsival** [comment]
Most of the failing tests were fixed in #71385 but there are a few that will require further investigation and are possibly caused by missing functionality in Android PAL (#45741)
